File: test_api.js
const fs = require('fs');
const path = require('path');
const FormData = require('form-data');
const http = require('http');

async function testConversion() {
  const testFile = path.join(__dirname, 'python', 'test_input.csv');
  
  if (!fs.existsSync(testFile)) {
    console.log('Test file not found');
    return;
  }
  
  const form = new FormData();
  form.append('tool', 'csv-to-excel');
  form.append('options', '{}');
  form.append('file', fs.createReadStream(testFile), 'test.csv');
  
  const options = {
    hostname: 'localhost',
    port: 3001,
    path: '/api/data-convert',
    method: 'POST',
    headers: form.getHeaders(),
  };
  
  const req = http.request(options, (res) => {
    console.log(`Status: ${res.statusCode}`);
    console.log('Headers:', res.headers);
    
    let data = '';
    res.on('data', chunk => data += chunk);
    res.on('end', () => {
      if (res.statusCode === 200) {
        console.log('✓ API call succeeded!');
        console.log(`Response size: ${data.length} bytes`);
      } else {
        console.log('✗ API call failed');
        console.log('Response:', data.substring(0, 500));
      }
    });
  });
  
  req.on('error', (err) => {
    console.log(`✗ Error: ${err.message}`);
  });
  
  form.pipe(req);
}

testConversion();

File: test_api.py
"""Test PDF to Text conversion via API"""
import requests
import os

# Test file and output
pdf_path = "test.pdf"
output_dir = "test_results"

# Create output directory
os.makedirs(output_dir, exist_ok=True)

# API endpoint
api_url = "http://localhost:3000/api/convert"

# Prepare files and data
files = {
    'image': ('test.pdf', open(pdf_path, 'rb'), 'application/pdf'),
    'config': (None, '''{
        "from_format": "pdf",
        "to_format": "txt",
        "options": {
            "language": "eng"
        }
    }''', 'application/json')
}

print("Testing PDF → TXT conversion via API...")
print(f"File: {pdf_path}")

try:
    response = requests.post(api_url, files=files, timeout=60)
    print(f"✓ Response Status: {response.status_code}")
    print(f"✓ Content Type: {response.headers.get('Content-Type', 'N/A')}")
    
    if response.status_code == 200:
        output_path = os.path.join(output_dir, "output.txt")
        with open(output_path, 'wb') as f:
            f.write(response.content)
        
        print(f"✓ Output saved: {output_path}")
        print(f"✓ Output size: {len(response.content)} bytes")
        
        # Show preview
        text_content = response.content.decode('utf-8', errors='ignore')
        preview = text_content[:300] if text_content else "[Empty output]"
        print("\nOutput preview:")
        print(preview)
    else:
        print(f"✗ Error: {response.status_code}")
        print(f"Response: {response.text[:500]}")
        
except Exception as e:
    print(f"✗ Error: {str(e)}")
